webapp/src/auth.ts

```diff
@@ -0,0 +1,218 @@
+import * as core from "./core";
+import * as data from "./data";
+
+import U = pxt.Util;
+
+/**
+ * Virtual API keys
```

Comment:
What do you mean by "virtual api" in this file?

Reply:
The "virtrual api" is the mechanism for injecting state into React components. It is defined in data.ts, here: https://github.com/microsoft/pxt/blob/541d9c867beaed9029aefb2cf65a3c4bcb1efaba/webapp/src/data.ts#L32